aws-opentelemetry-distro/src/amazon/opentelemetry/distro/patches/_instrumentation_patch.py

```diff
@@ -20,6 +20,12 @@ def apply_instrumentation_patches() -> None:
 
     Where possible, automated testing should be run to catch upstream changes resulting in broken patches
     """
+    if _is_installed("gevent"):
+        # pylint: disable=import-outside-toplevel
+        # Delay import to only occur if monkey patch is needed (e.g. gevent is used to run application).
+        from gevent import monkey
+
+        monkey.patch_ssl()
```

Comment:
The [gevent.monkey](https://www.gevent.org/api/gevent.monkey.html) doc recommends using `patch_all()`.
>The primary interface to this is the [patch_all()](https://www.gevent.org/api/gevent.monkey.html#gevent.monkey.patch_all) function, which performs all the available patches. It accepts arguments to limit the patching to certain modules, but most programs should use the default values as they receive the most wide-spread testing, and some monkey patches have dependencies on others.

What's the reason we are not using the `patch_all()` here?

*Just thinking out loud here:* 
I don't trust the `gevent.monkey` module mainly because 1) it is a 3rd party code which I don't know much about. 2) It is used in our distro for a very specific user base. So I'm thinking how can we make this safe and possible for users to get out of a possible bad situation caused by this code without having to wait for us to fix and release.

Maybe we can introduce an environment variable (something like `AWS_GEVENT_PATCH_MODULES`??) to control the behavior of this patching? The env variable can accept `all`, `none`, or a comma separated list of modules to patch which can be passed into the [`patch_all()` method](https://www.gevent.org/api/gevent.monkey.html#gevent.monkey.patch_all). By default we patch all if gevent is installed but still give users control to either completely disable this patch or patch select modules.

I know this will make the code a little bit complex and we will have to maintain an an extra environment variable, but it provides improved user experience.

Reply:
@thpierce any thoughts on this?

Reply:
`patch_ssl()` is a sub-command under `patch_all()`. patch_all() also patches some other libraries such as os, sys, time, etc ([link](https://www.gevent.org/_modules/gevent/monkey.html#patch_all)) We're using `patch_ssl()` to minimize the scope of the patch.